airflow-core/src/airflow/ui/src/layouts/Details/TaskStreamFilter.tsx

```diff
@@ -0,0 +1,184 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Portal, Text, VStack } from "@chakra-ui/react";
+import { useTranslation } from "react-i18next";
+import { FiChevronDown, FiFilter } from "react-icons/fi";
+
+import { Button } from "src/components/ui";
+import { Menu } from "src/components/ui/Menu";
+
+type TaskStreamFilterProps = {
+  readonly currentTaskId?: string;
+  readonly filterRoot: string | undefined;
+  readonly includeDownstream: boolean;
+  readonly includeUpstream: boolean;
+  readonly onIncludeDownstreamChange: (include: boolean) => void;
+  readonly onIncludeUpstreamChange: (include: boolean) => void;
+  readonly onFilterRootChange: (root: string | undefined) => void;
+};
+
+export const TaskStreamFilter = ({
+  currentTaskId,
+  filterRoot,
+  includeDownstream,
+  includeUpstream,
+  onIncludeDownstreamChange,
+  onIncludeUpstreamChange,
+  onFilterRootChange,
+}: TaskStreamFilterProps) => {
+  const { t: translate } = useTranslation(["components", "dag"]);
+  const isCurrentTaskTheRoot = currentTaskId === filterRoot;
```

Comment:
We can just pull the taskId from useParams() directly in this component instead of drilling it down as a prop.